core/include/scipp/core/dataset.h

```diff
@@ -1100,6 +1100,13 @@ class SCIPP_CORE_EXPORT DataArray {
     setCoord(dim, Variable(coord));
   }
 
+  void setMask(const std::string &name, Variable mask) {
+    m_holder.setMask(name, std::move(mask));
+  }
+  void setMask(const std::string &name, const VariableConstProxy &mask) {
+    setMask(name, Variable(mask));
+  }
+
```

Comment:
I think I had avoided adding these to `DataArray`, also for labels, attrs, etc. I recommend to use `array.masks().set(name, mask)` instead. That is also more in-line with how we write Python code.